new-log-viewer/src/contexts/StateContextProvider.tsx

```diff
@@ -138,36 +145,71 @@ const StateContextProvider = ({children}: StateContextProviderProps) => {
     const logEventNumRef = useRef(logEventNum);
     const numPagesRef = useRef<number>(STATE_DEFAULT.numPages);
     const pageNumRef = useRef<Nullable<number>>(STATE_DEFAULT.pageNum);
+    const receivedNumChunksRef = useRef<number>(0);
 
     const mainWorkerRef = useRef<null|Worker>(null);
 
     const handleMainWorkerResp = useCallback((ev: MessageEvent<MainWorkerRespMessage>) => {
         const {code, args} = ev.data;
+
+        // Create a file blob and push the data inside
+        const blob = new Blob();
+        const url = URL.createObjectURL(blob);
+        const numChunks = Math.ceil(numEvents / EXPORT_LOGS_CHUNK_SIZE);
```

Comment:
Shall we make this a React ref?

Reply:
Now we have two variables related to log exports and should better be stored in a `ref`, shall we consider making a class to encapsulate those variables for better context maintenance?